src/components/AddAssignment.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import Cookies from 'js-cookie';
+import { SERVER_URL } from '../constants.js';
+
+class AddAssignment extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { name: '', dueDate: '', course: '', redirect: false };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    event.preventDefault();
+    switch (event.target.id) {
+      case 'name':
+        this.setState({ name: event.target.value });
+        break;
+      case 'dueDate':
+        this.setState({ dueDate: event.target.value });
+        break;
+      case 'course':
+        this.setState({ course: event.target.value });
+        break;
+      default:
+        break;
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.postAssignment();
+  }
+
+  postAssignment = () => {
+    const token = Cookies.get('XSRF-TOKEN');
+    fetch(`${SERVER_URL}/course/${this.state.course}/assignment`,
+      {
+        method: 'POST',
+        headers: {
+          'X-XSRF-TOKEN': token,
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          'assignmentName': this.state.name,
+          'dueDate': new Date(this.state.dueDate).toISOString()
+        })
+      })
+      .then((response) => response.json())
+      .then((responseData) => {
+        console.log(responseData);
+        this.setState({ ...this.state, redirect: true });
+      })
+      .catch(err => console.error(err));
+  };
+
+  render() {
+    const { redirect } = this.state;
+
+    if (redirect) {
+      return <Redirect to='/'/>;
+    }
+
+    return (
+      <div align="left">
+        <h4 style={{ marginLeft: '3rem' }}>Add Assignment</h4>
+        <form onSubmit={this.handleSubmit} style={{
```

Comment:
Maybe for this you can also do a return due date and course?

Reply:
I don't understand this one. What do you mean? 🤔